File: backend/src/main/java/ru/chsergeig/shoppy/dao/AccountRoleRepository.java
package ru.chsergeig.shoppy.dao;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;
import org.jooq.Configuration;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Repository;
import ru.chsergeig.shoppy.jooq.enums.AccountRole;
import ru.chsergeig.shoppy.jooq.tables.daos.AccountsRolesDao;

import java.util.List;

import static ru.chsergeig.shoppy.jooq.Tables.ACCOUNTS;
import static ru.chsergeig.shoppy.jooq.Tables.ACCOUNTS_ROLES;

@Repository
public class AccountRoleRepository
        extends AccountsRolesDao {

    public AccountRoleRepository(
            @Autowired Configuration configuration
    ) {
        super(configuration);
    }

    /**
     * Get account roles by its login
     */
    @NotNull
    public List<AccountRole> fetchRolesByLogin(@Nullable String login) {
        return ctx().select(ACCOUNTS_ROLES.ROLE)
                .from(ACCOUNTS)
                .leftJoin(ACCOUNTS_ROLES).on(ACCOUNTS.ID.eq(ACCOUNTS_ROLES.ACCOUNT_ID))
                .where(ACCOUNTS.LOGIN.eq(login))
                .fetchInto(AccountRole.class);
    }

    public void deleteByAccountId(Integer id) {
        if (id == null) {
            return;
        }
        ctx().deleteFrom(ACCOUNTS_ROLES)
                .where(ACCOUNTS_ROLES.ACCOUNT_ID.eq(id))
                .execute();
    }
}

File: backend/src/main/kotlin/ru/chsergeig/shoppy/dao/AccountRoleRepository.kt
package ru.chsergeig.shoppy.dao

import org.jooq.DAO
import org.jooq.Record2
import ru.chsergeig.shoppy.jooq.enums.AccountRole
import ru.chsergeig.shoppy.jooq.tables.pojos.AccountsRoles
import ru.chsergeig.shoppy.jooq.tables.records.AccountsRolesRecord

interface AccountRoleRepository :
    DAO<AccountsRolesRecord, AccountsRoles, Record2<Int?, AccountRole?>> {

    fun fetchRolesByLogin(
        login: String?
    ): List<AccountRole>

    fun fetchByAccountId(
        ids: List<Int>
    ): MutableList<AccountsRoles>

    fun deleteByAccountId(
        id: Int?
    )
}
